### PR TITLE
resolves #4

### LINKED ISSUE #4: Should support markdown-it-container
Should also support other plugins that require options

### PR DESCRIPTION
**Pull request by sudsy**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -40,7 +40,12 @@ module.exports = function (source) {
 
   if (plugins) {
     plugins.forEach(function (plugin) {
-      parser.use(plugin);
+		 if(plugin.push){
+			 // allow array of options to be passed
+			 parser.use.apply(parser, plugin);
+		 }else{
+			parser.use(plugin);
+		 }
```
Comment (unindented) [inline index.js:48]: Can you please indent using two spaces, like the rest of the file?

File: package.json [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
   "author": {
     "name": "Daniel Perez Alvarez",
     "email": "unindented@gmail.com",
     "url": "http://unindented.org/"
   },
+  "contributors": [{
+	  "name": "Ben Sudbury",
+	  "email": "ben@lighthouselogic.com.au",
+	  "url": "http://www.lighthouselogic.com/"
+	}],
```
Comment (unindented) [inline package.json:14]: Please indent using two spaces.

File: test/loader.js [hunk 3]
```diff
@@ -44,6 +45,16 @@ module.exports.test = {
     test.done();
   },
 
+  'allows plugin options': function (test) {
+	 var sourceWithContainer = sourceWithPlugins + "\n ::: contained \n I am Contained \n :::"
+	 var context = mock({use: [[container, "contained"], subscript, superscript]});
```
Comment (unindented) [inline test/loader.js:50]: Please indent using two spaces, like the rest of the file.

File: index.js [hunk 4]
```diff
@@ -40,7 +40,12 @@ module.exports = function (source) {
 
   if (plugins) {
     plugins.forEach(function (plugin) {
-      parser.use(plugin);
+		 if(plugin.push){
```
Comment (unindented) [inline index.js:43]: Please use `Array.isArray`, I think the intention will be clearer.

File: index.js [hunk 5]
```diff
@@ -40,7 +40,12 @@ module.exports = function (source) {
...
+			 // allow array of options to be passed
+			 parser.use.apply(parser, plugin);
+		 }else{
```
Comment (unindented) [inline index.js:46]: Can you leave spaces around the `if` and `else` clauses, like in the rest of the file?

**Comment by unindented** [comment]
Left a few comments, but looks good. If you correct them and squash into one commit, I'll merge it.

Thank you!

**Comment by unindented** [comment]
@sudsy do you have the time to make the requested changes?

**Comment by sudsy** [comment]
I'm sure I'll get to them eventually. Probably not for the next couple of weeks.

**Pull request closed without merging**